base/strings/unicode.jl

```diff
@@ -532,8 +532,8 @@ iscntrl(c::AbstractChar) = c <= '\x1f' || '\x7f' <= c <= '\u9f'
     ispunct(c::AbstractChar) -> Bool
 
 Tests whether a character belongs to the Unicode general category Punctuation, i.e. a
-character whose category code begins with 'P'.
-
+character whose category code begins with 'P'.\n
+*Note*: This behavior is different from the ispunct function in C which checks for printable characters that are not alphanumeric or whitespace, based on the ASCII.
```

Comment:
```suggestion
!!! note
    This behavior is different from the `ispunct` function in C.
```

- Use the "note" syntax so this will be rendered as a note in docs and docstrings
- Make note shorter. This is a matter of opinion; IMO docstrings should almost exclusively focus on on how folks should use the function in Julia. In rare cases where behavior differs significantly and/or subtly from a very widespread standard (e.g. C) it makes sense to point that out, but I'd still only do so briefly. The point of this note is to give folks familiar with that standard (who presumably already know how `ispunct` is defined in C) a heads-up that they are dealing with something different. If folks want to know how C defines `ispunct` they should consult resources about C.

Reply:
thanks for the feedback! I have implemented the requested changes :)